src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/Subscription.cs

```diff
@@ -22,7 +22,7 @@ protected Subscription(IJsonRpcDuplexClient jsonRpcDuplexClient)
         public string Id { get; }
         public abstract string Type { get; }
         public IJsonRpcDuplexClient JsonRpcDuplexClient { get; }
-        private Channel<Action> SendChannel { get; } = Channel.CreateUnbounded<Action>(new UnboundedChannelOptions() { SingleReader = true });
```

Comment:
Would making the channel static - shared between subscriptions - fix the problem?
I think regardless we could/should make it static, there is no point in each subscription having own channel and it may cause problems?

Reply:
Note that this essentially reverts the changes introduced in #3458 so we should keep an eye on those old tickets (#3457 and #3444).

Reply:
Why making the channel static didn't fix the problem? Then it is shared by all subscriptions and should have make all messages sequential?

Reply:
Or if we want to keep the lock we don't need channels at all?

Reply:
> Or if we want to keep the lock we don't need channels at all?

Channels were introduced in #3458 to ensure ordering of messages. I believe that we could remove them and it should not affect the behavior due to locking, but I didn't want to introduce such change in this PR.

Reply:
Need to double check, but I don't think that when two thread is waiting for a lock, there is a guarentee that they are released in order.

Reply:
Also you can have communication on websocket that isn't subscriptions but normal request-response. I think we could downscale to having 1 channel per all subscriptions though, which would make it cheaper.